Fig.Core/Settings.cs

```diff
@@ -154,45 +177,117 @@ public T Bind<T>(bool requireAll = true, string prefix = null) where T : new()
         /// <typeparam name="T"></typeparam>
         public void Bind<T>(T target, bool requireAll = true, string prefix = null) where T : new()
         {
-            prefix = ((prefix ?? String.Empty).Trim() == String.Empty)
+            var errors = new List<string>();
+
+            var result = GetBindResult(target, requireAll, prefix, false, errors);
+
+            if (result.Errors.Any()) throw new ConfigurationException(errors);
+        }
+
+        private BindResult<T> GetBindResult<T>(bool requireAll = true, string prefix = null, bool preload = false, List<string> errors = null) where T : new()
+        {
+            var t = new T();
+            return GetBindResult(t, requireAll, prefix, preload, errors);
+        }
+
+        private BindResult<T> GetBindResult<T>(T target, bool requireAll = true, string prefix = null, bool preload = false, List<string> errors = null) where T : new()
```

Comment:
I think the name GetBindResult is a bit misleading. It's not merely getting the result, it's doing the actual binding. Maybe `BindProperties` or some other qualifier to distinguish from the other `Bind` overloads.